store/src/main/java/com/nytimes/android/external/store3/pipeline/PipelinePersister.kt

```diff
@@ -1,157 +1,138 @@
 package com.nytimes.android.external.store3.pipeline
 
+import com.nytimes.android.external.cache3.CacheBuilder
 import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.channels.Channel
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.channelFlow
-import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.collectIndexed
-import kotlinx.coroutines.flow.collectLatest
-import kotlinx.coroutines.flow.consumeAsFlow
-import kotlinx.coroutines.flow.emitAll
-import kotlinx.coroutines.flow.flatMapLatest
-import kotlinx.coroutines.flow.flow
-import kotlinx.coroutines.flow.mapNotNull
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.channels.*
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.channels.actor
+
 
 class PipelinePersister<Key, Input, Output>(
-    private val fetcher: PipelineStore<Key, Input>,
-    private val reader: (Key) -> Flow<Output?>,
-    private val writer: suspend (Key, Input) -> Unit,
-    private val delete: (suspend (Key) -> Unit)? = null
+        private val fetcher: PipelineStore<Key, Input>,
+        private val reader: (Key) -> Flow<Output?>,
+        private val writer: suspend (Key, Input) -> Unit,
+        private val delete: (suspend (Key) -> Unit)? = null
 ) : PipelineStore<Key, Output> {
-    @ExperimentalCoroutinesApi
-    @Suppress("UNCHECKED_CAST")
-    override fun stream(request: StoreRequest<Key>): Flow<StoreResponse<Output>> {
-        return when {
-            request.shouldSkipCache(CacheType.DISK) -> fetchSkippingCache(request)
-            else -> diskNetworkCombined(request)
-        }
-    }
+    private val responseFlows = CacheBuilder
```

Comment:
i guess we don't want to use cache here.
Keeping these around when there are no requesters might be OK (not sure necessary but possible); though possibility of losing it because cache is full sounds bad.

Reply:
yup we can figure this out once we get tests passing, should be easy to flip to something like keytracker or expire on last subscriber